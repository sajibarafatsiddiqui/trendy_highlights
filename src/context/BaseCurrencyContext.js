import React from 'react';

const BaseCurrencyContext = React.createContext();
export default BaseCurrencyContext;
