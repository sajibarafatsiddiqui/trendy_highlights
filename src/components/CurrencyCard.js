import React, { useContext } from 'react';
import { PropTypes } from 'prop-types';
import { Card } from 'react-bootstrap';
import style from 'styles/CurrencyCard.module.css';
import { NavLink } from 'react-router-dom';
import BaseCurrencyContext from 'context/BaseCurrencyContext';
import cx from 'classnames';
import { IoArrowForwardCircleOutline } from 'react-icons/io5';

const CurrencyCard = ({ currency, rate, isDark }) => {
  const { baseCurrency } = useContext(BaseCurrencyContext);
  console.log(isDark);
  return (
    <Card style={{ backgroundColor: isDark ? '#CF4277' : '#e94987', padding: '0' }} className={cx(style[currency], style.Card)}>
      <Card.Body style={{ margin: '0' }}>
        <NavLink style={{ color: 'white', textDecoration: 'none' }} to={`/details/${baseCurrency}-${currency}`} state={rate}>
          <IoArrowForwardCircleOutline style={{
            top: '0', right: '0', position: 'absolute',
          }}
          />
          <Card.Title />
          <Card.Text>{ currency }</Card.Text>
          <Card.Footer style={{
            bottom: '0', right: '0', position: 'absolute',
          }}
          >
            { Number(rate).toFixed(4) }
          </Card.Footer>

        </NavLink>

      </Card.Body>
    </Card>

  );
};
CurrencyCard.propTypes = {
  currency: PropTypes.string.isRequired,
  rate: PropTypes.number.isRequired,
  isDark: PropTypes.bool.isRequired,
};
export default CurrencyCard;
