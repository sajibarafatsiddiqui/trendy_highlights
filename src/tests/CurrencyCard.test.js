import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import CurrencyCard from 'components/CurrencyCard';
import { Provider } from 'react-redux';
import store from 'redux/store';
import BaseCurrencyContext from 'context/BaseCurrencyContext';
import { BrowserRouter as Router } from 'react-router-dom';

describe('Tests for <Conversion /> component', () => {
  const currency = (
    <CurrencyCard
      currency="EUR"
      rate={100.023232}
      isDark={false}
    />
  );
  const baseCurrency = 'USD';

  const wrap = (component) => (
    <Provider store={store}>
      <BaseCurrencyContext.Provider value={{ baseCurrency }}>
        <Router>
          {component}
        </Router>
      </BaseCurrencyContext.Provider>
    </Provider>
  );
  it('Should render the content correctly', () => {
    render(wrap(currency));
    expect(screen.getByText('EUR')).not.toBeNull();
    expect(screen.getByText('100.0232')).not.toBeNull();
  });

  it('Should match the snapshot', () => {
    expect(renderer.create(wrap(currency)).toJSON()).toMatchSnapshot();
  });
});
