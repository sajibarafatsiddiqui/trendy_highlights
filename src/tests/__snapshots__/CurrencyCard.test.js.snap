// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <Conversion /> component Should match the snapshot 1`] = `
<div
  className="EUR Card card"
  style={
    Object {
      "backgroundColor": "#e94987",
      "padding": "0",
    }
  }
>
  <div
    className="card-body"
    style={
      Object {
        "margin": "0",
      }
    }
  >
    <a
      className=""
      href="/details/USD-EUR"
      onClick={[Function]}
      style={
        Object {
          "color": "white",
          "textDecoration": "none",
        }
      }
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
            "position": "absolute",
            "right": "0",
            "top": "0",
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M262.62 336L342 256l-79.38-80m68.35 80H170"
          fill="none"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="32"
        />
        <path
          d="M256 448c106 0 192-86 192-192S362 64 256 64 64 150 64 256s86 192 192 192z"
          fill="none"
          strokeMiterlimit="10"
          strokeWidth="32"
        />
      </svg>
      <div
        className="card-title h5"
      />
      <p
        className="card-text"
      >
        EUR
      </p>
      <div
        className="card-footer"
        style={
          Object {
            "bottom": "0",
            "position": "absolute",
            "right": "0",
          }
        }
      >
        100.0232
      </div>
    </a>
  </div>
</div>
`;
